### PR TITLE
[docs] gh-101273: Expand enum documentation on limitations

### LINKED ISSUE #101273: Expand on enum limitations
# Documentation

Encountered the python/mypy#14494 issue and it was suggested that we include it on the docs. This issue is an atttempt to expand on the documentation for this particular issue.

### Linked PRs
* gh-101274

### PR DESCRIPTION
**Pull request by samypr100**
Closes #101273

* Issue: gh-101273

### CODE REVIEW

**Review by ethanfurman — COMMENTED** [review]
I'm not sure this needs to be in the docs; I am sure that this is not the right section for them to be in.  I'll try to review the enum docs this week to see if there is a better spot for this warning.

**Comment by samypr100** [comment]
@ethanfurman To expand a little on the motivation, the codebase that I originally encountered this had something close to 

```python
class ElementType(StrEnum):
    title = "..."
    comment = "..."
    author = "..."
    # ...
```

It didn't occur to me at first that the errors I was seeing in mypy (mypy was being added after the fact) were due to `StrEnum` inheriting from `str`, hence why I also thought that it might be a good idea to add a warning/note to the docs.

File: Misc/NEWS.d/next/Documentation/2023-01-23-17-55-15.gh-issue-101273.Ae-Hed.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Expand :mod:`enum` documentation on limitations when shadowing a built-in.
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Documentation/2023-01-23-17-55-15.gh-issue-101273.Ae-Hed.rst]: We don't need news for docs :)

File: Doc/library/enum.rst [hunk 2]
```diff
@@ -925,3 +925,17 @@ Notes
        >>> from enum import IntEnum
        >>> class MyIntEnum(IntEnum):
        ...     __str__ = IntEnum.__str__
+
+   Additionally, it's possible to end up shadowing a built-in function from
```
Comment (sobolevn) [inline Doc/library/enum.rst]: `.title` is not a builtin function. It is a method of `str`.

File: Doc/library/enum.rst [hunk 3]
```diff
@@ -925,3 +925,17 @@ Notes
...
+   a derived baseclass such as :meth:`str.title` or :meth:`int.bit_count`,
```
Comment (sobolevn) [inline Doc/library/enum.rst]: Let's add one more example here. It feels like 2 is not enough :)

File: Doc/library/enum.rst [hunk 4]
```diff
@@ -925,3 +925,17 @@ Notes
...
+   which can cause unexpected issues at runtime.
+
+   Consider the following example::
+
+       >>> from enum import StrEnum
+       >>> class MyStrEnum(StrEnum):
+       ...     title = "RED"
+       >>> MyStrEnum.title.title()
```
Comment (sobolevn) [inline Doc/library/enum.rst:938]: I think it is worth showing that other `str` methods work fine: `.uppercase()` for example.

Comment (ethanfurman) [reply Doc/library/enum.rst:938]: Not needed.  `.uppercase` will stop working as soon as an `uppercase` member is created -- which is the point of the doc addition.

File: Misc/NEWS.d/next/Documentation/2023-01-23-17-55-15.gh-issue-101273.Ae-Hed.rst (hunk 1 above)
Comment (samypr100) [reply Misc/NEWS.d/next/Documentation/2023-01-23-17-55-15.gh-issue-101273.Ae-Hed.rst]: Removed in afde363e70c626697e04e141359413caff0d708e

File: Doc/library/enum.rst (hunk 2 above)
Comment (samypr100) [reply Doc/library/enum.rst]: Gasp, I had it as method 😢 originally. Dunno why I changed it. Fixed in afde363e70c626697e04e141359413caff0d708e

File: Doc/library/enum.rst (hunk 3 above)
Comment (samypr100) [reply Doc/library/enum.rst]: Added another example in afde363e70c626697e04e141359413caff0d708e

**Comment by ethanfurman** [comment]
Closing in favor of #103596.  Thank you, @samypr100 for the encouragement to properly fix this behavior.

**Pull request closed without merging**